rust/foxglove/src/websocket/ws_protocol/client/playback_control_request.rs

```diff
@@ -0,0 +1,201 @@
+use bytes::{Buf, BufMut};
+
+use crate::websocket::ws_protocol::{BinaryMessage, ParseError};
+
+use super::BinaryOpcode;
+
+#[doc(hidden)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[repr(u8)]
+pub enum PlaybackCommand {
+    Play = 0,
+    Pause = 1,
+}
+
+impl TryFrom<u8> for PlaybackCommand {
+    type Error = u8;
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(Self::Play),
+            1 => Ok(Self::Pause),
+            _ => Err(value),
+        }
+    }
+}
+
+#[doc(hidden)]
+/// A request to control playback from the client
+#[derive(Debug, Clone, PartialEq)]
+pub struct PlaybackControlRequest {
+    /// Playback state
+    pub playback_command: PlaybackCommand,
+    /// Playback speed
+    pub playback_speed: f32,
+    /// Seek playback time in nanoseconds (only set if a seek has been performed)
+    pub seek_time: Option<u64>,
+    /// Unique string identifier, used to indicate that a PlaybackState is in response to a particular request from the client
+    pub request_id: String,
+}
+
+impl<'a> BinaryMessage<'a> for PlaybackControlRequest {
+    // Message layout:
+    //   opcode (1 byte)
+    // + playback_command (1 byte)
+    // + playback_speed (4 bytes)
+    // + had_seek (1 byte)
+    // + seek_time (8 bytes)
+    // + request_id_len (4 bytes)
+    // + request_id
+    fn parse_binary(mut data: &'a [u8]) -> Result<Self, ParseError> {
+        if data.len() < 1 + 4 + 1 + 8 {
```

Comment:
Tack on 4 more for `request_id_len`

```suggestion
        if data.len() < 1 + 4 + 1 + 8 + 4 {
```